scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1605,33 +1586,16 @@ public void removeThirdPartyCredential(String key) throws NotConnectedException,
 
     @Override
     @ImmediateService
-    public Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks, boolean running,
-            boolean pending, boolean finished, int offset, int limit, SortSpecifierContainer sortParams)
+    public Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks,
+            Set<TaskStatus> statusFilter, int offset, int limit, SortSpecifierContainer sortParams)
             throws NotConnectedException, PermissionException {
-        RestPageParameters params = new RestPageParameters(frontendState,
-                                                           "getTaskStates",
-                                                           from,
-                                                           to,
-                                                           mytasks,
-                                                           running,
-                                                           pending,
-                                                           finished,
-                                                           offset,
-                                                           limit,
-                                                           taskTag,
-                                                           sortParams);
-        Page<TaskState> pTasks;
-        pTasks = dbManager.getTaskStates(params.getFrom(),
-                                         params.getTo(),
-                                         params.getTag(),
-                                         params.getOffset(),
-                                         params.getLimit(),
-                                         params.getUserName(),
-                                         params.isPending(),
-                                         params.isRunning(),
-                                         params.isFinished(),
-                                         params.getSortParams());
-        return pTasks;
+
+        String userName = null;
+        if (mytasks) {
+            userName = frontendState.checkPermission("frontendState", "You do not have permission to use frontendState")
```

Comment:
This permission does not exist

I see "getTaskStates" in the code

Reply:
NICE